backend/user-service/psql.ts

```diff
@@ -0,0 +1,44 @@
+import { Pool } from 'pg';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const pool = new Pool({
+    host     : process.env.PSQL_HOSTNAME,
+    database : process.env.PSQL_DATABASE,
+    user     : process.env.PSQL_USERNAME,
+    password : process.env.PSQL_PASSWORD,
+    port     : parseInt(process.env.PSQL_PORT || '5432')
+});
+
+const createUserTableQueryIfNotExist = `
+    DO $$ 
+    BEGIN
+        IF NOT EXISTS (
+          SELECT 1
+          FROM information_schema.tables
+          WHERE table_name = 'users'
+        ) THEN
+          CREATE TABLE users (
+              user_id SERIAL PRIMARY KEY,
+              email VARCHAR(255) NOT NULL,
+              name VARCHAR(255) NOT NULL,
+              major VARCHAR(255) NOT NULL,
+              course VARCHAR(255),
```

Comment:
We don't need to store course for now, but one important attribute to store is the assignments, you need to have foreign key referencing the Assignment table on one to many relation.

Reply:
Oh wait I want to revert this. I forgot that we agree that for current implementation, we assume every user with student role is from the same class, so when there are assignments available, everyone should get it. But I think it is still good to have an assignments attribute here.